### PR TITLE
Improve release process in GH workflows

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Improve release process in GH workflows
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | -
| Sponsor company   | -
| How to test?      | -

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    if: github.event.release.prerelease == false
+    runs-on: ubuntu-latest
+    needs: create_release_zip
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Download release asset
+        run: |
+          gh release download ${{ github.event.release.tag_name }} -R PrestaShop/autoupgrade -p ${{ github.event.repository.name }}-${{ github.event.release.tag_name }}.zip
```
Comment (Quetzacoalt91) [inline .github/workflows/publish.yml:44]:
```suggestion
          gh release download ${{ github.event.release.tag_name }} -R ${{ github.event.repository.full_name }} -p ${{ github.event.repository.name }}-${{ github.event.release.tag_name }}.zip
```

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Prepare the zip
+        run: |
+          cd ../
+          zip -r ${{ github.event.repository.name }}-${{ github.event.release.tag_name }}.zip ${{ github.event.repository.name }} -x '*.git*' 
+          mv ${{ github.event.repository.name }}-${{ github.event.release.tag_name }}.zip ./${{ github.event.repository.name }}
+          cd ./${{ github.event.repository.name }}
+
+      - name: Publish the zip on release
+        run: |
+          gh release upload ${{ github.event.release.tag_name }} ${{ github.event.repository.name }}-${{ github.event.release.tag_name }}.zip -R /PrestaShop/autoupgrade
```
Comment (Quetzacoalt91) [inline .github/workflows/publish.yml:29]:
```suggestion
          gh release upload ${{ github.event.release.tag_name }} ${{ github.event.repository.name }}-${{ github.event.release.tag_name }}.zip -R /${{ github.event.repository.full_name }}
```

**Review by Quetzacoalt91 — COMMENTED** [review]
I'm in favor of relying on variables whenever possible for re-usability.

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
Before merging, we need to remove the autoupgrade folder from https://github.com/PrestaShop/PrestaShop-modules/

**Comment by M0rgan01** [comment]
@Quetzacoalt91 https://github.com/PrestaShop/PrestaShop-modules/pull/450

**Pull request merged**